koku/api/report/queries.py

```diff
@@ -765,22 +766,27 @@ def _ranked_list(self, data_list):
             if other is None:
                 other = copy.deepcopy(data)
             rank = data.get('rank')
-            if rank <= self._limit:
+            if rank <= self._limit and ranked_dict.get(rank) is None:
```

Comment:
I think we may actually need `row_number` on top of this change. As an example let's say we get back the ranks 1, 2, 2, 3, 4, 5, 5, 6 with limit set to 5. I think we'd display 1, the first 2, 3, 4, the first 5, and 6 while putting the second 2 and second 5 into the others category.